apps/ebclfsa-demo/Readme.md

```diff
@@ -0,0 +1,129 @@
+
+
+## Communication
+The demo uses two shared memory segments:
+ * proxycom: This is used to communicate from a low integrity application to a high integrity application.
+   In the demo it is partitioned into multiple chunks, one for each running high integrity application.
+   The high integrity applications register themselves in this memory with a UUID, that is used to find the correct partition by the low integrity application.
+ * hicom: This is used to communicate between high integrity applications.
+   It cannot be accessed by low integrity applications
+
+Messages are passed between applications using the shared memory and a ringbuffer.
+
+The following sequence of messages is sent:
+
+1. li_demo sends a message over the proxycom shared memory to hi_main.
+2. The message is received by hi_main, manipulated and send to hi_forward over the hicom shared memory.
+3. hi_forward then sends the message back to hi_demo using proxycom.
+
+## Watchdog
+There is a watchdog process running as part of the hypervisor.
+The watchdog has to be triggered regularly by the hi integrity application.
+Since there is only one watchdog available, aggregation of watchdog signals is the responsibility of the (first) high integrity application.
+Therefor the second high integrity application (hi_forward) signals a watchdog to the first high integrity application (hi_main) using the hicom shared memory.
+Then hi_main can trigger the watchdog.
+
+The watchdog leaves the system some time to completely boot up.
+For qemu this timeout is rather long (~50s), because qemu performance can vary a lot.
+A message is printed to the console, when this timeout is expired and the watchdog is active:
+
+    safehm_gpio_supervisor: msg received mr[0]=3855 eol
+
+
+From now on the watchdog must be triggered periodically, if it is not triggered any more the following message is printed. This can be triggered by running `li_demo -s`, which tells li_main to stop triggering the watchdog.
+
+    SafeHM: (ERROR) Heartbeat not received from Client with id = 12
+    SafeHM: (ERROR) safe_state (exit_code=-1)
+
+The watchdog cannot be restarted and the system has to be rebooted.
+
+NOTE: For qemu this has no further effect.
+
+
+## Not allowed syscalls
+Not all syscalls are allowed in ebclfsa high integrity applications.
+In the qemu setup calling a prohibited syscall does not have any effect apart from a message on the console.
+On a real safety system calling a prohibited syscall will trigger a transition to the safe more.
+
+To test executing a not allowed syscall `li_demo -p` can be executed, which calls `ioctl(0, 0)` and yields the following message:
+
+    SDK:handler_do_el0_svc_pre: syscall __NR_ioctl (29) is not allowed
+
+NOTE: The current state of the syscalls whitelist does not reflect the final list of allowed syscalls!
+
+## Interacting with the hypervisor (advanced)
+
+The hypervisor console can be accessed using the 'CTRL+E .' key combination.
```

Comment:
@antonjahn I wouldn't call this "THE" hypervisor console. What you can access here is the console muxer and I don't even know if it adds any value to be able to access it in the current state (i.e. as long as the hi vm has no way of ineracting with it using the console)